homeassistant/components/smarty.py

```diff
@@ -0,0 +1,224 @@
+        }),
+    },
+    extra=vol.ALLOW_EXTRA)
+
+RPM = 'rpm'
+SIGNAL_UPDATE_SMARTY = 'smarty_update'
+
+INPUT_REGS = {
+    'IR_CURRENT_SYSTEM_STATE': {
+        'ADDR': 1,
+        'NAME': 'Current system state',
+        'MULTIPLIER': 1,
+        'STATE': {
+            0: 'Stand-by',
+            1: 'Building protection',
+            2: 'Economy',
+            3: 'Comfort',
+            4: 'Emergency run',
+            5: 'Preparing',
+            6: 'Opening dampers',
+            7: 'Boost',
+            8: 'Cooling heaters',
+            9: 'Closing dampers',
+            10: 'Night Cooling',
+            11: 'Critical alarm',
+            12: 'Fire alarm',
+            13: 'Heat exchanger frost protection',
+            14: 'Change filters',
+            15: 'Room RH 3 days average is lower than 30%. Limiting speed',
+            16: 'DX cooler defrosting',
+            17: 'Fire damper testing'
+        },
+        'UNIT_OF_MESUREMENT': None
+    },
+    'IR_CURRENT_SYSTEM_MODE': {
+        'ADDR': 15,
+        'NAME': 'Current system mode',
+        'MULTIPLIER': 1,
+        'STATE': {
+            0: 'Stand-by',
+            1: 'Building protection',
+            2: 'Economy',
+            3: 'Comfort',
+            4: 'Boost'
+        },
+        'UNIT_OF_MESUREMENT': None
+    },
+    'IR_SUPPLY_AIR_TEMPERATURE': {
+        'ADDR': 18,
+        'NAME': 'T1-Supply air temperature',
+        'MULTIPLIER': 0.1,
+        'STATE': {},
+        'UNIT_OF_MESUREMENT': TEMP_CELSIUS
+    },
+    'IR_EXTRACT_AIR_TEMPERATURE': {
+        'ADDR': 19,
+        'NAME': 'T2-Extract air temperature',
+        'MULTIPLIER': 0.1,
+        'STATE': {},
+        'UNIT_OF_MESUREMENT': TEMP_CELSIUS
+    },
+    'IR_OUTDOOR_AIR_TEMPERATURE': {
+        'ADDR': 21,
+        'NAME': 'Outdoor air temperature',
+        'MULTIPLIER': 0.1,
+        'STATE': {},
+        'UNIT_OF_MESUREMENT': TEMP_CELSIUS
+    },
+    'IR_SUPPLY_FAN_SPEED_RPM': {
+        'ADDR': 55,
+        'NAME': 'Supply fan spreed RPM',
+        'MULTIPLIER': 1,
+        'STATE': {},
+        'UNIT_OF_MESUREMENT': RPM
+    },
+    'IR_EXTRACT_FAN_SPEED_RPM': {
+        'ADDR': 56,
+        'NAME': 'Extract fan spreed RPM',
+        'MULTIPLIER': 1,
+        'STATE': {},
+        'UNIT_OF_MESUREMENT': RPM
+    }
+}
+
+HOLDING_REGS = {
+    'HR_USER_CONFIG_CURRENT_SYSTEM_MODE': {
+        'ADDR': 1,
+        'NAME': 'Current system mode',
+        'STATE': {
+            'Stand-by': 0,
+            'Building protection': 1,
+            'Economy': 2,
+            'Confort': 3
+        }
+    }
+}
+
+
+def setup(hass, config):
+    """Set up the smarty environment."""
+    conf = config[DOMAIN]
+
+    name = conf.get(CONF_NAME)
+    modbus_slave = conf.get(CONF_SLAVE)
+
+    _LOGGER.debug("name: %s, modbus_slave: %s", name, modbus_slave)
+
+    smarty = Smarty(hass, name, modbus_slave)
+    hass.data[DATA_SMARTY] = smarty
+
+    # Load platforms
+    hass.async_create_task(
```

Comment:
This async api. We can't use async functions from sync context. Just call `load_platform` directly.